### ISSUE TITLE
tcp/uct_p2p_mix_test.mix_10000/0 hangs on hpc_test_node_upstream

### ISSUE THREAD

**Issue by dmitrygx** [openucx/ucx#3769]
```
19:57:06 Note: This is test shard 4 of 4.
19:57:06 Note: Randomizing tests' orders with a seed of 21469 .
```
```
20:04:27 [----------] 1 test from tcp/uct_p2p_mix_test
20:04:27 [ RUN      ] tcp/uct_p2p_mix_test.mix_10000/0
20:19:26 /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/test_helpers.cc:46: Failure
20:19:26 Failed
20:19:26 Connection timed out - abort testing
20:19:27 [hpc-test-node-upstream:56335:0:56335] Caught signal 6 (Aborted: tkill(2) or tgkill(2))
20:21:14 ==== backtrace (tid:  56335) ====
20:21:14  0 0x00000000000f8921 __epoll_wait_nocancel()  :0
20:21:14  1 0x000000000005b68f ucs_event_set_wait()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../src/ucs/sys/event_set.c:152
20:21:14  2 0x000000000001a423 uct_tcp_iface_progress()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../src/uct/tcp/tcp_iface.c:162
20:21:14  3 0x0000000000630f5a uct_test::entity::flush()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../src/ucs/datastruct/callbackq.h:211
20:21:14  4 0x00000000005fd92d uct_p2p_mix_test::run()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/uct/test_p2p_mix.cc:159
20:21:14  5 0x0000000000560916 ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/test.cc:276
20:21:14  6 0x000000000054b503 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/gtest-all.cc:3562
20:21:14  7 0x000000000053f97d testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/gtest-all.cc:3635
20:21:14  8 0x000000000053fa4c testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/gtest-all.cc:3812
20:21:14  9 0x000000000053fbaf testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/gtest-all.cc:3930
20:21:14 10 0x0000000000544547 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/gtest-all.cc:5802
20:21:14 11 0x000000000054484b testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/gtest-all.cc:5719
20:21:14 12 0x00000000004eb543 main()  /scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/gtest.h:20059
20:21:14 13 0x0000000000021c05 __libc_start_main()  ???:0
20:21:14 14 0x000000000052a104 _start()  ???:0
20:21:14 =================================
```
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/label=hpc-test-node-upstream,worker=3/10793/console

**Comment by dmitrygx** [comment]
Fixed in #3859

**Comment by dmitrygx** [comment]
reproduced on hpc-test-node-upstream02 under valgrind:
```
[2019-08-21T13:49:48.466Z] [     INFO ] Using random seed of 19452
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_IB_RX_QUEUE_LEN=512
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_IB_RX_BUFS_GROW=512
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_MM_RX_BUFS_GROW=128
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_IB_TX_QUEUE_LEN=128
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_IB_TX_BUFS_GROW=64
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_RC_TX_CQ_LEN=256
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_CM_TIMEOUT=600ms
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_TCP_TX_BUFS_GROW=512
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_TCP_RX_BUFS_GROW=512
[2019-08-21T13:49:48.466Z] [     INFO ]  Setting for valgrind: UCX_TCP_RX_SEG_SIZE=16k
[2019-08-21T13:49:48.721Z] Note: This is test shard 1 of 4.
[2019-08-21T13:49:48.721Z] Note: Randomizing tests' orders with a seed of 23183 .
...
[2019-08-21T14:00:23.775Z] [----------] 1 test from tcp/uct_p2p_mix_test
[2019-08-21T14:00:23.775Z] [ RUN      ] tcp/uct_p2p_mix_test.mix_10000/0
[2019-08-21T14:15:30.561Z] /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/common/test_helpers.cc:49: Failure
[2019-08-21T14:15:30.561Z] Failed
[2019-08-21T14:15:30.561Z] Connection timed out - abort testing
[2019-08-21T14:15:30.561Z] [hpc-test-node-upstream-02:25033:0:25033] Caught signal 6 (Aborted: tkill(2) or tgkill(2))
[2019-08-21T14:16:51.967Z] ==== backtrace (tid:  25033) ====
[2019-08-21T14:16:51.967Z]  0 0x00000000000f8921 __epoll_wait_nocancel()  :0
[2019-08-21T14:16:51.967Z]  1 0x0000000000060103 ucs_event_set_wait()  /scrap/jenkins/workspace/ucx-5/contrib/../src/ucs/sys/event_set.c:196
[2019-08-21T14:16:51.967Z]  2 0x000000000001c694 uct_tcp_iface_progress()  /scrap/jenkins/workspace/ucx-5/contrib/../src/uct/tcp/tcp_iface.c:192
[2019-08-21T14:16:51.967Z]  3 0x00000000006438da uct_test::entity::flush()  /scrap/jenkins/workspace/ucx-5/contrib/../src/ucs/datastruct/callbackq.h:211
[2019-08-21T14:16:51.967Z]  4 0x000000000060f6dd uct_p2p_mix_test::run()  /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/uct/test_p2p_mix.cc:159
[2019-08-21T14:16:51.967Z]  5 0x0000000000571cd6 ucs::test_base::run()  /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/common/test.cc:276
[2019-08-21T14:16:51.967Z]  6 0x0000000000559433 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/common/gtest-all.cc:3562
[2019-08-21T14:16:51.967Z]  7 0x000000000054d8ad testing::Test::Run()  /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/common/gtest-all.cc:3635
[2019-08-21T14:16:51.967Z]  8 0x000000000054d97c testing::TestInfo::Run()  /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/common/gtest-all.cc:3812
[2019-08-21T14:16:51.967Z]  9 0x000000000054dadf testing::TestCase::Run()  /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/common/gtest-all.cc:3930
[2019-08-21T14:16:51.967Z] 10 0x0000000000552477 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/common/gtest-all.cc:5802
[2019-08-21T14:16:51.967Z] 11 0x000000000055277b testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/common/gtest-all.cc:5719
[2019-08-21T14:16:51.967Z] 12 0x00000000004f5313 main()  /scrap/jenkins/workspace/ucx-5/contrib/../test/gtest/common/gtest.h:20059
[2019-08-21T14:16:51.967Z] 13 0x0000000000021c05 __libc_start_main()  ???:0
[2019-08-21T14:16:51.967Z] 14 0x0000000000538034 _start()  ???:0
[2019-08-21T14:16:51.967Z] =================================
[2019-08-21T14:16:51.967Z] Sending notification to dmitrygla@mellanox.com
[2019-08-21T14:17:59.622Z] [hpc-test-node-upstream-02:25033:0:25033] Process frozen...
```
http://hpc-master.lab.mtl.com:8080/blue/organizations/jenkins/ucx/detail/ucx/1424/pipeline/112

**Comment by yosefe** [comment]
@dmitrygx is this still relevant?

**Comment by dmitrygx** [comment]
> @dmitrygx is this still relevant?

@yosefe I can't reproduce this issue. Hope it can be fixed by #4163
Let's close and re-open it when it appears again
